### ISSUE TITLE
Kill hasattr(sys, 'exc_clear') as it's there since 2.3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4309]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#4309|
|Type|enhancement|
|Created|2010-02-23 20:13:56Z|
|Branch|https://github.com/twisted/twisted/tree/sys-exc_clear-4309|

Twisted no longer supports 2.3 any more. sys.exc_clear check (hasattr) should go away from twisted/trial/runner.py

<details><summary>Searchable metadata</summary>

```
trac-id__4309 4309
type__enhancement enhancement
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_sys_exc_clear_4309 branches/sys-exc_clear-4309
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1266956036000000 1266956036000000
changetime__1267376563000000 1267376563000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess commented|
|-|-|

(In [28541]) Branching to 'sys-exc_clear-4309'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [28542]) Remove unnecessary hasattr check.

refs [#4309](https://github.com/twisted/twisted/issues/4309)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess removed owner|
|-|-|

[indeed](http://docs.python.org/library/sys.html#sys.exc_clear).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @jesstess|
|-|-|

Cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set status to `closed`|
|-|-|

(In [28575]) Merge sys-exc_clear-4309

Author: jesstess
Reviewer: therve
Fixes: [#4309](https://github.com/twisted/twisted/issues/4309)

Removed hasattr(sys, 'exc_clear') checks in runner.py as exc_clear
exists in all Pythons supported by Twisted (Pythons >= 2.3).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
